yax/modules/identify_informative_hits.py

```diff
@@ -0,0 +1,217 @@
+                        (total_informative_hits/100*min_relative_hits))
+
+    informative_tax_ids = []
+    for tax_id, num_hits in hit_tax_ids.items():
+        if num_hits >= hit_threshold:
+            informative_tax_ids.append(tax_id)
+
+    return informative_tax_ids
+
+
+def get_hit_taxa(hit_taxa_lines):
+    """Builds dictionary from read_ids and the tax_ids it hit
+
+    A set of all tax_ids it hit is constructed for each read_id.
+
+    parameters
+    ----------
+        hit_taxa_lines : list
+            all lines from mg_wrapper output generated in an alignment module
+
+    Returns
+    -------
+        dict
+            read_id key to list of tax_ids value where the read was observed
+            to hit the tax_id in an alignment step
+
+    """
+    hit_taxa = {}
+
+    for line in hit_taxa_lines:
+        this_line = line.rstrip().split(":")
+        this_read_id = this_line[1]
+        these_hits = set()
+        for mm_distance in this_line[1:]:
+            for tax_id_hit in mm_distance:
+                these_hits.add(tax_id_hit)
+
+        hit_taxa[this_read_id] = list(these_hits)
+
+    return hit_taxa
+
+
+def get_hit_taxids(hit_data, hit_taxa, lca_dist):
+    """Counts the numer of times each tax_id was hit in alignment
+
+    For each tax_id the number of reads that informatively aligned to it are
+    counted.
+
+    Parameters
+    ----------
+        hit_data : object
+            containg tree and data recording final informative output
+        hit_taxa : dict
+            read_id key to list of hit tax_ids value
+        lca_dist : Int
+            number of edges to be traversed when determining lowest common
+            ancestor
+
+    Returns
+    -------
+        dict
+            tax_id key to number of hits to it value
+
+    """
+    hit_tax_ids = {}
+    total_informative_hits = 0
+    for read_id in hit_taxa:
+        these_tax_ids_hit = hit_taxa[read_id]
+        if len(these_tax_ids_hit) == 1:
+            total_informative_hits += 1
+            hit_tax_ids = increment_tax_id_hits(
+                hit_tax_ids, these_tax_ids_hit[0])
+
+        else:
+            informative = True
+            ancestor_pool = set(hit_data.tax_id_data[these_tax_ids_hit[0]].
+                                parents[-lca_dist:])
+            for tax_id_hit in these_tax_ids_hit[1:]:
+                this_ancestor_pool = set(hit_data.tax_id_data[tax_id_hit].
+                                         parents[-lca_dist:])
+                if this_ancestor_pool & ancestor_pool:
+                    ancestor_pool |= this_ancestor_pool
+                else:
+                    break
+                    informative = False
+            if informative:
+                total_informative_hits += 1
+                hit_tax_ids = increment_tax_id_hits(
+                    hit_tax_ids, these_tax_ids_hit[0])
+
+    return hit_tax_ids, total_informative_hits
+
+
+def increment_tax_id_hits(ids, id_to_increment):
```

Comment:
Surely there is a better way... Look into `collections.Counter`, it may be applicable